### PR TITLE
influxdb (on nixos): reduce closure size by 99.99% (and a bit)

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

The influxdb nixos module pulls in ```collectd``` in order to fetch a single file which pulls in a number of other things (java, gtk and loads more).

This PR creates a separate derivation for the just the data bits and makes it configurable for influxdb.

Closure size before: 897M
Closure size after: 36K

My question is - should we instead change the collectd derivation to properly use multiple outputs and then stick the data bits in there instead of making the separate ```collectd-data``` derivation?

@joachifm @fpletz 

###### Things done

Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers.

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/modules/services/databases/influxdb.nix [hunk 1]
```diff
@@ -68,7 +68,7 @@ let
 
     collectd = [{
       enabled = false;
-      typesdb = "${pkgs.collectd}/share/collectd/types.db";
+      typesdb = "${cfg.collectdPackage}/share/collectd/types.db";
```
Comment (joachifm) [inline nixos/modules/services/databases/influxdb.nix]: Is there a use-case for *not* using the data-only variant? If not, maybe just do that, at least for the initial version.

Comment (peterhoeg) [inline nixos/modules/services/databases/influxdb.nix]: Good point, I'll do that. There is really only one collectd package anyway.

Comment (peterhoeg) [inline nixos/modules/services/databases/influxdb.nix]: Any thoughts on the multiple outputs alternative?

Comment (joachifm) [inline nixos/modules/services/databases/influxdb.nix]: No strong opinion on it. Conceptually I think multiple-outputs is cleaner but not sure if there are any practical benefits in this case.

Is it possible to deploy `collectd` without the data? If so, that'd be a good case for moving the data to a separate output.  Another thing to consider is that without hardlinking optimization you'd waste quite a bit of space if both the data-only and the full collectd paths end up being referenced by a gc root.

**Comment by fpletz** [comment]
I would prefer to use multiple outputs here if collectd still works or we can make it work with the data being in another output. But this approach is also fine for now. :+1:

**Comment by peterhoeg** [comment]
I've updated the PR to just reference ```collectd-data``` directly. If everyone is OK with this I'll go ahead and merge it (unless I'm beaten to it...)

**Comment by volth** [comment]
Besides your ```collectd/data.nix``` there are
1. [```libcollectdclient```](https://github.com/NixOS/nixpkgs/blob/56904d7c423f2b13b37fbd29f39bbb4b52bc7824/pkgs/development/libraries/libcollectdclient/default.nix)
2. [```lighttpd/collectd```](https://github.com/NixOS/nixpkgs/blob/c2576ac2d37882709be41227202a1dc691b22192/nixos/modules/services/web-servers/lighttpd/collectd.nix#L29)

which could also become outputs of ```collectd```

**Pull request merged**